### ISSUE TITLE
Telescope fails to create a popup window

### ISSUE THREAD

**Issue by alexfertel** [nvim-telescope/telescope.nvim#1105]
### Description

When I run `find_files` or `live_grep` it crashes displaying this:

```
E5108: Error executing lua ...re/nvim/plugged/telescope.nvim/lua/telescope/pickers.lua:302: attempt to call field 'create' (a nil value) 
```

**Expected Behavior**

It should open the popup window.

**Actual Behavior**

It doesn't open the popup window.

### Details

<details><summary>Reproduce</summary>

1. `nvim -nu test.vim`
2. :Telescope live_grep or git_commits

</details>

<details><summary>Environment</summary>

- nvim --version output: 

```
NVIM v0.6.0-dev+132-g7d2233fad
Build type: Release
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/cc -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT -O2 -DNDEBUG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/home/alex/alex/software/neovim/build/config -I/home/alex/alex/software/neovim/src -I/home/alex/alex/software/neovim/.deps/usr/include -I/usr/include -I/home/alex/alex/software/neovim/build/src/nvim/auto -I/home/alex/alex/software/neovim/build/include
Compiled by root@alice

Features: +acl +iconv +tui
```

- Operating system: Ubuntu 18.04 
- Telescope commit: latest

</details>

<details><summary>Configuration</summary>
<p>

I copied the snippet below to a `test.vim` file and ran `nvim -nu test.vim`.

```viml
set nocompatible hidden laststatus=2

if !filereadable('/tmp/plug.vim')
  silent !curl --insecure -fLo /tmp/plug.vim
    \ https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim
endif

source /tmp/plug.vim
call plug#begin('/tmp/plugged')
Plug 'nvim-lua/plenary.nvim'
Plug 'nvim-telescope/telescope.nvim'
call plug#end()

autocmd VimEnter * PlugClean! | PlugUpdate --sync | close
lua require('telescope').setup()
```

**Comment by Conni2461** [comment]
I can't reproduce this, which is weird.
Can you tell me what `:lua print(vim.inspect(require("plenary.popup")))` returns?

**Comment by clason** [comment]
Also, have you removed the now obsolete `popup.nvim` plugin?

**Comment by alexfertel** [comment]
> Also, have you removed the now obsolete `popup.nvim` plugin?

Yeap, just did and it's the same.



> :lua print(vim.inspect(require("plenary.popup")))

```
{                                                                                                                                                                                                         
  _hidden = {},                                                                                                                                                                                           
  _pos_map = {                                                                                                                                                                                            
    botleft = "SW",                                                                                                                                                                                       
    botright = "SE",                                                                                                                                                                                      
    topleft = "NW",                                                                                                                                                                                       
    topright = "NE"                                                                                                                                                                                       
  },                                                                                                                                                                                                      
  popup_create = <function 1>,                                                                                                                                                                            
  show = <function 2>                                                                                                                                                                                     
} 
```

Thank you for having patience! :pray:

**Comment by clason** [comment]
Something is off, then: I have
```
{
  _callbacks = {},
  _hidden = {},
  _pos_map = {
    botleft = "SW",
    botright = "SE",
    topleft = "NW",
    topright = "NE"
  },
  create = <function 1>,
  execute_callback = <function 2>
}
```
So you must have stale plugins lying around.

**Comment by alexfertel** [comment]
> Something is off, then: I have
> 
> ```
> {
>   _callbacks = {},
>   _hidden = {},
>   _pos_map = {
>     botleft = "SW",
>     botright = "SE",
>     topleft = "NW",
>     topright = "NE"
>   },
>   create = <function 1>,
>   execute_callback = <function 2>
> }
> ```
> 
> So you must have stale plugins lying around.

Let me try to reinstall from scratch once again.

**Comment by gikeymarcia** [comment]
> you could also check here: `lua print(vim.inspect(vim.api.nvim_list_runtime_paths()))` if the correct plenary plugin is loaded. The plugin you currently load, looks out of date.

I get the same return from `:lua print(vim.inspect(require("plenary.popup")))` as @alexfertel 

```
{
  _hidden = {}, 
  _pos_map = { 
    botleft = "SW", 
    botright = "SE", 
    topleft = "NW", 
    topright = "NE"
    },
    popup_create = <function 1>, 
    show = <function 2>
}
```
When I run `lua print(vim.inspect(vim.api.nvim_list_runtime_paths()))` I get back a lot

```lua
{
    "/home/mikey/.config/nvim",
    "/tmp/plugged/plenary.nvim",
    "/tmp/plugged/telescope.nvim",
    "/etc/xdg/xdg-awesome/nvim",
    "/etc/xdg/nvim",
    "/home/mikey/.local/share/nvim/site",
    "/home/mikey/.local/share/nvim/site/pack/packer/start/vim-rooter",
    "/home/mikey/.local/share/nvim/site/pack/packer/start/telescope.nvim",
    "home/mikey/.local/share/nvim/site/pack/packer/start/popup.nvim",
    "home/mikey/.local/share/nvim/site/pack/packer/start/plenary.nvim",
    "/home/mikey/.local/share/nvim/site/pack/packer/start/packer.nvim",
    "/home/mikey/.local/share/nvim/site/pack/packer/start/nvim-web-devicons",
    "/home/mikey/.local/share/nvim/site/pack/packer/start/nvim-treesitter",
    "/home/mikey/.local/share/nvim/site/pack/packer/start/nvim-tree.lua",
    "/home/mikey/.local/share/nvim/site/pack/packer/start/nvim-lspconfig",
    "/home/mikey/.local/share/nvim/site/pack/packer/start/nvim-compe",
    "/home/mikey/.local/share/nvim/site/pack/packer/start/formatter.nvim",
    "/home/mikey/.local/share/nvim/site/pack/packer/start/astronauta.nvim",
    "/usr/share/awesome/nvim/site",
    "/home/mikey/.local/share/flatpak/exports/share/nvim/site",
    "/var/lib/flatpak/exports/share/nvim/site",
    "/usr/local/share/nvim/site",
    "/usr/share/nvim/site",
    "/var/lib/snapd/desktop/nvim/site",
    "/tmp/.mount_nvimOX2xJE/usr/share/nvim/runtime",
    "/tmp/.mount_nvimOX2xJE/usr/share/nvim/runtime/pack/dist/opt/matchit",
    "/tmp/.mount_nvimOX2xJE/usr/lib/nvim",
    "/home/mikey/.local/share/nvim/site/pack/packer/start/nvim-treesitter/after",
    "/home/mikey/.local/share/nvim/site/pack/packer/start/nvim-compe/after",
    "/var/lib/snapd/desktop/nvim/site/after",
    "/usr/share/nvim/site/after",
    "/usr/local/share/nvim/site/after",
    "/var/lib/flatpak/exports/share/nvim/site/after",
    "/home/mikey/.local/share/flatpak/exports/share/nvim/site/after",
    "/usr/share/awesome/nvim/site/after",
    "/home/mikey/.local/share/nvim/site/after",
    "/etc/xdg/nvim/after",
    "/etc/xdg/xdg-awesome/nvim/after",
    "/home/mikey/.config/nvim/after"
}

```

**Comment by clason** [comment]
There's your problem right there: you're mixing plugins (via vim-plug) and packages. Note how you have _two_ different telescope plugins in your path?

**Comment by alexfertel** [comment]
So now something even weirder is happening to me. I just went crazy and deleted all of the installed plugins everywhere I could find them, including `vim-plug`.

I ran

```bash
sh -c 'curl -fLo "${XDG_DATA_HOME:-$HOME/.local/share}"/nvim/site/autoload/plug.vim --create-dirs \
       https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim'
```

Which puts the `plug.vim` file here `/home/alex/.local/share/nvim/site/autoload/plug.vim` and here `/home/alex/.vim/autoload/plug.vim`.

But now I cannot get `PlugInstall` to work.

Running `lua print(vim.inspect(vim.api.nvim_list_runtime_paths()))` (or equivalently `:echo &rtp` outputs:

```lua
{ "/home/alex/.config/nvim", "/etc/xdg/xdg-ubuntu/nvim", "/etc/xdg/nvim", "/home/alex/.local/share/nvim/site", "/usr/share/ubuntu/nvim/site", "/home/alex/.local/share/flatpak/exports/share/nvim/site", "
/var/lib/flatpak/exports/share/nvim/site", "/usr/local/share/nvim/site", "/usr/share/nvim/site", "/var/lib/snapd/desktop/nvim/site", "/usr/local/share/nvim/runtime", "/usr/local/share/nvim/runtime/pack/
dist/opt/matchit", "/usr/local/lib/nvim", "/var/lib/snapd/desktop/nvim/site/after", "/usr/share/nvim/site/after", "/usr/local/share/nvim/site/after", "/var/lib/flatpak/exports/share/nvim/site/after", "/
home/alex/.local/share/flatpak/exports/share/nvim/site/after", "/usr/share/ubuntu/nvim/site/after", "/home/alex/.local/share/nvim/site/after", "/etc/xdg/nvim/after", "/etc/xdg/xdg-ubuntu/nvim/after", "/
home/alex/.config/nvim/after" }
```

Which includes `"/home/alex/.local/share/nvim/site"`, so I am out of ideas 😢

**Comment by clason** [comment]
...and that's a _mess_ of non-default paths...

**Comment by clason** [comment]
(`.vim` is irrelevant for neovim)

**Comment by alexfertel** [comment]
> (`.vim` is irrelevant for neovim)

Yeah, said it just in case.

**Comment by Conni2461** [comment]
@gikeymarcia i think your output was gather with your plugins still loaded and the minimal rc loaded. From your other report i can see that you ran `nvim -Nu`. its `nvim -nu` lowercase `n`. Can you try it again only minimal rc

**Comment by gikeymarcia** [comment]
> There's your problem right there: you're mixing plugins (via vim-plug) and packages. Note how you have _two_ different telescope plugins in your path? And `popup.nvim` still installed as a package?

Thank you! I also figured out where my issue came from. When beginning to switch from vim to NeoVim I first messed with [LunarVim](https://github.com/LunarVim/LunarVim) and used his [installer](https://github.com/LunarVim/LunarVim/blob/rolling/utils/installer/install.sh) to get up and running. After a bit of playing around I decided I wanted to build my own config.

What I didn't consider was all of the stuff left behind in `~/.local/share/nvim/site` from packer installs. After deleting the folder everything is working again.

**TLDR**, Build your own configs so you know what is in them.

@Conni2461 Good catch. Changed to `-n` and Telescope is working again. Thanks for helping me troubleshoot and being a FOSS hero. I hope this interaction can help others who made my mistake.

**Comment by alexfertel** [comment]
I also figured out mine! Thanks, everyone. It seems that I was using some stale version of plenary somehow. I just did a fresh install, including `vim-plug` (Don't ask me how I fixed the issue above), and everything is working fine. 😁

**Comment by tami5** [comment]
> > There's your problem right there: you're mixing plugins (via vim-plug) and packages. Note how you have _two_ different telescope plugins in your path? And `popup.nvim` still installed as a package?
> 
> 
> 
> Thank you! I also figured out where my issue came from. When beginning to switch from vim to NeoVim I first messed with [LunarVim](https://github.com/LunarVim/LunarVim) and used his [installer](https://github.com/LunarVim/LunarVim/blob/rolling/utils/installer/install.sh) to get up and running. After a bit of playing around I decided I wanted to build my own config.
> 
> 
> 
> What I didn't consider was all of the stuff left behind in `~/.local/share/nvim/site` from packer installs. After deleting the folder everything is working again.
> 
> 
> 
> **TLDR**, Build your own configs so you know what is in them.
> 
> 
> 
> @Conni2461 Good catch. Changed to `-n` and Telescope is working again. Thanks for helping me troubleshoot and being a FOSS hero. I hope this interaction can help others who made my mistake.

from someone who rewritten configuration from scratch a number of times, I really appreciate something like LunarVim. It just takes the maintenance if architecture out of your plate.
> Build your own configs so you know what is in them.

agree, I'd add fork to it when you find will design and structure config. 


Maintaining neovim configuration is hard 😣. best advice is to strive for clean architecture, lazy loading and tag releases (yes discipline is required)
